File: DXDAL/LotCodeFormatSections.vb
Imports System
Imports DevExpress.Data.Filtering
Imports DevExpress.Xpo

Namespace SPGData
    Public Class LotCodeFormatSections
        Inherits AuditableBase

        Dim fLotCodeFormat As LotCodeFormats
        <Association("Format-Sections")>
        Public Property LotCodeFormat() As LotCodeFormats
            Get
                Return fLotCodeFormat
            End Get
            Set(value As LotCodeFormats)
                SetPropertyValue("LotCodeFormat", fLotCodeFormat, value)
            End Set
        End Property

        Dim fSectionName As String
        <Size(50)>
        Public Property SectionName() As String
            Get
                Return fSectionName
            End Get
            Set(value As String)
                SetPropertyValue("SectionName", fSectionName, value)
            End Set
        End Property

        Dim fSectionType As LotCodeSectionType
        Public Property SectionType() As LotCodeSectionType
            Get
                Return fSectionType
            End Get
            Set(value As LotCodeSectionType)
                SetPropertyValue("SectionType", fSectionType, value)
            End Set
        End Property

        Dim fSectionLength As Integer
        Public Property SectionLength() As Integer
            Get
                Return fSectionLength
            End Get
            Set(value As Integer)
                SetPropertyValue(Of Integer)("SectionLength", fSectionLength, value)
            End Set
        End Property

        Dim fSectionPosition As Integer
        Public Property SectionPosition() As Integer
            Get
                Return fSectionPosition
            End Get
            Set(value As Integer)
                SetPropertyValue(Of Integer)("SectionPosition", fSectionPosition, value)
            End Set
        End Property
        
        <Association("Section-Settings", GetType(LotCodeSectionSettings)), Aggregated()>
        Public ReadOnly Property SectionSettings() As XPCollection(Of LotCodeSectionSettings)
            Get
                Return GetCollection(Of LotCodeSectionSettings)("SectionSettings")
            End Get
        End Property
        Public Sub New()
            MyBase.New()
            ' This constructor is used when an object is loaded from a persistent storage.
            ' Do not place any code here.			
        End Sub

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
            ' This constructor is used when an object is loaded from a persistent storage.
            ' Do not place any code here.			
        End Sub

        Public Overrides Sub AfterConstruction()
            MyBase.AfterConstruction()
            ' Place here your initialization code.
        End Sub

        Public Shadows Class FieldsClass
            Inherits XPLiteObject.FieldsClass
            Public Sub New()
                MyBase.New()
            End Sub
            Public Sub New(ByVal propertyName As String)
                MyBase.New(propertyName)
            End Sub
            Public ReadOnly Property Oid() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("Oid"))
                End Get
            End Property
            Public ReadOnly Property LotCodeFormat() As LotCodeFormats.FieldsClass
                Get
                    Return New LotCodeFormats.FieldsClass(GetNestedName("LotCodeFormat"))
                End Get
            End Property
            Public ReadOnly Property SectionName() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("SectionName"))
                End Get
            End Property
            Public ReadOnly Property SectionType() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("SectionType"))
                End Get
            End Property
            Public ReadOnly Property SectionLength() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("SectionLength"))
                End Get
            End Property
            Public ReadOnly Property SectionPosition() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("SectionPosition"))
                End Get
            End Property
        End Class
        Private Shared _fields As FieldsClass
        Public Shared Shadows ReadOnly Property Fields() As FieldsClass
            Get
                If ReferenceEquals(_fields, Nothing) Then
                    _fields = New FieldsClass()
                End If
                Return _fields
            End Get
        End Property

    End Class
End Namespace

File: DXDAL/LotCodeFormatSections.cs
using DevExpress.Data.Filtering;
using DevExpress.Xpo;

namespace DXDAL
{
    namespace SPGData
    {
        public class LotCodeFormatSections : AuditableBase
        {
            private LotCodeFormats fLotCodeFormat;
            [Association("Format-Sections")]
            public LotCodeFormats LotCodeFormat
            {
                get => fLotCodeFormat;
                set => SetPropertyValue("LotCodeFormat", ref fLotCodeFormat, value);
            }

            private string fSectionName;
            [Size(50)]
            public string SectionName
            {
                get => fSectionName;
                set => SetPropertyValue("SectionName", ref fSectionName, value);
            }

            private LotCodeSectionType fSectionType;
            public LotCodeSectionType SectionType
            {
                get => fSectionType;
                set => SetPropertyValue("SectionType", ref fSectionType, value);
            }

            private int fSectionLength;
            public int SectionLength
            {
                get => fSectionLength;
                set => SetPropertyValue<int>("SectionLength", ref fSectionLength, value);
            }

            private int fSectionPosition;
            public int SectionPosition
            {
                get => fSectionPosition;
                set => SetPropertyValue<int>("SectionPosition", ref fSectionPosition, value);
            }

            [Association("Section-Settings", typeof(LotCodeSectionSettings))]
            [Aggregated()]
            public XPCollection<LotCodeSectionSettings> SectionSettings => GetCollection<LotCodeSectionSettings>("SectionSettings");
            public LotCodeFormatSections() : base()
            {
            }

            public LotCodeFormatSections(Session session) : base(session)
            {
            }

            public override void AfterConstruction()
            {
                base.AfterConstruction();
            }

            public new class FieldsClass : PersistentBase.FieldsClass
            {
                public FieldsClass() : base()
                {
                }
                public FieldsClass(string propertyName) : base(propertyName)
                {
                }
                public OperandProperty Oid => new OperandProperty(GetNestedName("Oid"));
                public LotCodeFormats.FieldsClass LotCodeFormat => new LotCodeFormats.FieldsClass(GetNestedName("LotCodeFormat"));
                public OperandProperty SectionName => new OperandProperty(GetNestedName("SectionName"));
                public OperandProperty SectionType => new OperandProperty(GetNestedName("SectionType"));
                public OperandProperty SectionLength => new OperandProperty(GetNestedName("SectionLength"));
                public OperandProperty SectionPosition => new OperandProperty(GetNestedName("SectionPosition"));
            }
            private static FieldsClass _fields;
            public static new FieldsClass Fields
            {
                get
                {
                    if (ReferenceEquals(_fields, null))
                        _fields = new FieldsClass();
                    return _fields;
                }
            }
        }
    }
}
